apps/api/src/app/workflows-v2/usecases/build-step-issues/build-step-issues.usecase.ts

```diff
@@ -0,0 +1,258 @@
+import { Injectable } from '@nestjs/common';
+import { ControlValuesRepository } from '@novu/dal';
+import {
+  ContentIssue,
+  JSONSchemaDto,
+  StepContentIssueEnum,
+  StepIssuesDto,
+  UserSessionData,
+  StepTypeEnum,
+  WorkflowOriginEnum,
+  ControlValuesLevelEnum,
+} from '@novu/shared';
+import {
+  InstrumentUsecase,
+  TierRestrictionsValidateUsecase,
+  TierRestrictionsValidateCommand,
+  dashboardSanitizeControlValues,
+  PinoLogger,
+} from '@novu/application-generic';
+
+import _ from 'lodash';
+import Ajv, { ErrorObject } from 'ajv';
+import addFormats from 'ajv-formats';
+import { buildVariables } from '../../util/build-variables';
+import { BuildAvailableVariableSchemaCommand, BuildAvailableVariableSchemaUsecase } from '../build-variable-schema';
+import { BuildStepIssuesCommand } from './build-step-issues.command';
+
+@Injectable()
+export class BuildStepIssuesUsecase {
+  constructor(
+    private buildAvailableVariableSchemaUsecase: BuildAvailableVariableSchemaUsecase,
+    private controlValuesRepository: ControlValuesRepository,
+    private tierRestrictionsValidateUsecase: TierRestrictionsValidateUsecase,
+    private logger: PinoLogger
+  ) {}
+
+  @InstrumentUsecase()
+  async execute(command: BuildStepIssuesCommand): Promise<StepIssuesDto> {
+    const {
+      workflowOrigin,
+      user,
+      step: foundPersistedStep,
+      workflow: persistedWorkflow,
+      stepDto: step,
+      controlSchema,
+    } = command;
+
+    const variableSchema = await this.buildAvailableVariableSchemaUsecase.execute(
```

Comment:
No need for the word available :)

```suggestion
    const variableSchema = await this.buildVariableSchemaUsecase.execute(
```